.github/workflows/build-book.yaml

```diff
@@ -113,7 +113,7 @@ jobs:
         if: |
           (inputs.use_cached_environment != 'true'
           || steps.cache.outputs.cache-hit != 'true')
-          && steps.parse_config.outputs.execute_notebooks != 'binder'
+          && steps.parse_config.outputs.execute_notebooks == 'binder'
```

Comment:
@mgrover1 I think this is not the right fix for the problem you were having.

This line was meant to run this workflow stop *only if executing the notebooks on GitHub Actions*. In that case, we need to build an execution environment here.

For executing on binder, the `environment.yml` file is ignored and we just build a minimal JupyterBook + binderbot environment (see [this line](https://github.com/ProjectPythia/cookbook-actions/blob/36246f0920586405e6fd53753a7f5506af2e8285/.github/workflows/build-book.yaml#L102))

If I understand correctly, the problem you were trying to solve is the case where we execute on binder, but there is a change in the `environment.yml` file so a new image needs to be built on the binder. Is that correct? We'll need a different fix for that. 

Currently the call to binderbot is hard-wired to the `main` branch ([this line](https://github.com/ProjectPythia/cookbook-actions/blob/36246f0920586405e6fd53753a7f5506af2e8285/.github/workflows/build-book.yaml#L145)) so we always use the image associated with current main branch.

Reply:
Thanks for the clarification here @brian-rose and sorry for merging without getting more feedback first!